rust/main/utils/run-locally/src/server.rs

```diff
@@ -80,3 +82,69 @@ pub fn fetch_relayer_gas_payment_event_count() -> eyre::Result<u32> {
     .iter()
     .sum::<u32>())
 }
+
+pub fn send_insert_message_request() -> io::Result<insert_messages::ResponseBody> {
+    let runtime = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build();
+    runtime
+        .unwrap()
+        .block_on(async { send_insert_message_request_non_blocking().await })
+}
+
+async fn send_insert_message_request_non_blocking() -> io::Result<insert_messages::ResponseBody> {
+    let client = reqwest::Client::new();
+    let url =
+        Url::parse(&format!("http://0.0.0.0:{RELAYER_METRICS_PORT}/messages")).map_err(|err| {
+            println!("Failed to parse url: {err}");
+            io::Error::new(io::ErrorKind::InvalidInput, err.to_string())
+        })?;
+
+    let body = insert_messages::RequestBody {
+        messages: vec![
+            insert_messages::Message {
+                version: 0,
+                nonce: 10000,
+                origin: 9913371,
+                destination: 9913372,
+                sender: H256::from_low_u64_be(1000),
+                recipient: H256::from_low_u64_be(2000),
+                body: Vec::new(),
+                dispatched_block_number: 10000,
+            },
+            insert_messages::Message {
+                version: 0,
+                nonce: 10001,
+                origin: 9913371,
+                destination: 9913372,
+                sender: H256::from_low_u64_be(1000),
+                recipient: H256::from_low_u64_be(2000),
+                body: Vec::new(),
+                dispatched_block_number: 10001,
+            },
+        ],
+    };
+
+    let retry_response = client
```

Comment:
I wonder if `client` actually retries or the name of the variable is a remanence of copied code?